File: test_vulns/mock_vuln.py
"""
mock_vuln.py
Safe-for-testing example file that contains patterns commonly flagged by scanners.
Do NOT use this file in production. Only commit to a disposable test repository.
"""

# --- Hardcoded "secret" (intentionally benign placeholder) ---
API_KEY = "TEST_ONLY_DO_NOT_USE_0123456789"

def get_api_key():
    return API_KEY

def run_untrusted_expression(expr: str):
    # insecure pattern for static analysis tests (do not call)
    return eval(expr)

import subprocess
def run_command(cmd: str):
    subprocess.run(cmd, shell=True)

import pickle
def load_pickle_bytes(b: bytes):
    return pickle.loads(b)

import random
def insecure_token():
    choices = "abcdefghijklmnopqrstuvwxyz0123456789"
    return "".join(random.choice(choices) for _ in range(16))

import hashlib
def md5_hash(s: str):
    return hashlib.md5(s.encode()).hexdigest()

if __name__ == "__main__":
    print("This module is a static test artifact for security scanning only.")

File: test_vulns/MockVuln.java
/*
 MockVuln.java
 Safe-for-testing example class with patterns commonly flagged by static scanners.
 Do NOT use in production. Only commit to a disposable test repository.
*/

public class MockVuln {
    private static final String PASSWORD = "TEST_ONLY_DO_NOT_USE_PASSWORD_12345";
    public static String getPassword() { return PASSWORD; }

    public static String insecureMd5(String input) {
        try {
            java.security.MessageDigest md = java.security.MessageDigest.getInstance("MD5");
            byte[] digest = md.digest(input.getBytes("UTF-8"));
            StringBuilder sb = new StringBuilder();
            for (byte b : digest) sb.append(String.format("%02x", b));
            return sb.toString();
        } catch (Exception e) {
            return null;
        }
    }

    public static byte[] insecureAesEcb(byte[] key, byte[] plaintext) {
        try {
            javax.crypto.SecretKeySpec ks = new javax.crypto.SecretKeySpec(key, "AES");
            javax.crypto.Cipher cipher = javax.crypto.Cipher.getInstance("AES/ECB/PKCS5Padding");
            cipher.init(javax.crypto.Cipher.ENCRYPT_MODE, ks);
            return cipher.doFinal(plaintext);
        } catch (Exception e) {
            return null;
        }
    }

    public static void runShellCommand(String cmd) {
        try {
            Runtime.getRuntime().exec(new String[] { "/bin/sh", "-c", cmd });
        } catch (Exception e) {}
    }

    public static void main(String[] args) {
        System.out.println("MockVuln: static test artifact for security scanning only.");
    }
}
